packages/sdk/src/sdk/sdk/webpack.ts

```diff
@@ -479,4 +489,124 @@ export class RsdoctorWebpackSDK<
+    }
+    compressTextScripts = `${compressTextScripts} <script>window.${Constants.WINDOW_RSDOCTOR_TAG}.enableRoutes=${JSON.stringify(this.getClientRoutes())}</script>`;
+
+    htmlContent = htmlContent.replace('<body>', `<body>${compressTextScripts}`);
+
+    return htmlContent;
+  }
+
+  public inlineScriptsAndStyles(htmlFilePath: string) {
+    // Helper function to inline JavaScript files
+    function inlineScripts(basePath: string, scripts: string[]): string {
+      return scripts
+        .map((src) => {
+          const scriptPath = path.resolve(basePath, src);
+          try {
+            const scriptContent = fse.readFileSync(scriptPath, 'utf-8');
+            return `<script>${scriptContent}</script>`;
+          } catch (error) {
+            console.error(`Could not read script at ${scriptPath}:`, error);
+            return '';
+          }
+        })
+        .join('');
+    }
+
+    // Helper function to inline CSS files
+    function inlineCss(basePath: string, cssFiles: string[]): string {
+      return cssFiles
+        .map((href) => {
+          const cssPath = path.resolve(basePath, href);
+          try {
+            const cssContent = fse.readFileSync(cssPath, 'utf-8');
+            return `<style>${cssContent}</style>`;
+          } catch (error) {
+            console.error(`Could not read CSS at ${cssPath}:`, error);
+            return '';
+          }
+        })
+        .join('');
+    }
+
+    // Path to your HTML file
+    let htmlContent = fse.readFileSync(htmlFilePath, 'utf-8');
+
+    // Base path to the resources
+    const basePath = path.dirname(htmlFilePath);
+
+    // Extract scripts and links from the HTML
+    const scriptSrcs = Array.from(
+      htmlContent.matchAll(
+        /<script\s+defer="defer"\s+src=["'](.+?)["']><\/script>/g,
+      ),
+      (m) => m[1],
+    );
+    const cssHrefs = Array.from(
+      htmlContent.matchAll(/<link\s+href=["'](.+?)["']\s+rel="stylesheet">/g),
+      (m) => m[1],
+    );
+
+    // Remove all <script> tags
+    htmlContent = htmlContent.replace(
+      /<script\s+.*?src=["'].*?["']><\/script>/g,
+      '',
+    );
+
+    // Remove all <link rel="stylesheet"> tags
+    htmlContent = htmlContent.replace(
+      /<link\s+.*?rel=["']stylesheet["'].*?>/g,
+      '',
+    );
+
+    // Inline scripts and CSS
+    const inlinedScripts = inlineScripts(basePath, scriptSrcs);
+    const inlinedCss = inlineCss(basePath, cssHrefs);
+
+    const index = htmlContent.indexOf('</body>');
+    htmlContent =
+      htmlContent.slice(0, index) +
+      inlinedCss +
+      inlinedScripts +
+      htmlContent.slice(index);
+    htmlContent = this.addRsdoctorDataToHTML(this.getStoreData(), htmlContent);
+
+    // Output the processed HTML content
+    const outputFilePath = path.resolve(this.outputDir, 'rsdoctor-report.html');
+
+    fse.ensureFileSync(outputFilePath);
+
+    fse.writeFileSync(outputFilePath, htmlContent, {
```

Comment:
fse.ensureFileSync + fse.writeFileSync = fse.outputFileSync

Reply:
And prefer using async fs API

Reply:
done